lib/shopify_cli/theme/dev_server/hot-reload-tae.js

```diff
@@ -0,0 +1,173 @@
+(() => {
+  const APP_BLOCK = 'app-block', APP_EMBED_BLOCK = 'app-embed-block';
+
+  function connect() {
+    const eventSource = new EventSource('/hot-reload');
+
+    eventSource.onmessage = handleUpdate;
+
+    eventSource.onopen = () => console.log('[HotReload] SSE connected.');
+
+    eventSource.onclose = () => {
+      console.log('[HotReload] SSE closed. Attempting to reconnect...');
+
+      setTimeout(connect, 5000);
+    }
+
+    eventSource.onerror = () => eventSource.close();
+  }
+
+  function reloadMode() {
+    const namespace = window.__SHOPIFY_CLI_ENV__;
+    return namespace.mode;
+  }
+
+  function querySelectHotReloadElements(handle) {
+    // Gets all blocks (app and embed) with specified handle TODO: add app ID check here
+    const blocks = Array.from(document.querySelectorAll(`[data-block-handle$='${handle}']`));
+    if (blocks.length){
+      const queryString = 'shopify-section-template';
+      const is_section = blocks[0].closest(`[id^=${queryString}]`) !== null;
```

Comment:
this assumes we cannot have an app embed block and a regular app block with the same filename in the same project.

Reply:
That's a fair assumption :)

Reply:
That's a fair assumption :)